operator/pkg/central/values/translation/pvc_checker.go

```diff
@@ -0,0 +1,37 @@
+package translation
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// pvcExistenceChecker wraps up the information and function to detect
+type pvcExistenceChecker struct {
+	ctx         context.Context
+	client      ctrlClient.Client
+	obsoletePvc bool
+	nameSpace   string
+}
+
+func (c *pvcExistenceChecker) toObsolete() bool {
+	return c.obsoletePvc
+}
+
+func (c *pvcExistenceChecker) pvcExists(name string) bool {
+	key := ctrlClient.ObjectKey{Namespace: c.nameSpace, Name: name}
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := c.client.Get(c.ctx, key, pvc)
+	if !apiErrors.IsNotFound(err) {
```

Comment:
I think this would be cleaner to check if err != nil then check if it is not found. Also, is it safer to assume the none case? It seems safer to assume the other way which would not removed the ownership reference on the pvc

Reply:
This is to determine if we want to mount the pvc. It is in the reconcile to handle the references. If we mount a pvc that does not exists, central will hang.